src/overlays/actors/ovl_Bg_Dblue_Elevator/z_bg_dblue_elevator.c

```diff
@@ -5,68 +5,373 @@
+    PlayState* play = play2;
+    WaterBox* waterBox;
+    s32 bgId;
+
+    this->isWithinWaterBoxXZ =
+        WaterBox_GetSurfaceImpl(play, &play->colCtx, this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.z,
+                                &this->waterSurfaceYPos, &waterBox, &bgId);
+}
+
+void BgDblueElevator_SpawnRipplesAndSplashes(BgDblueElevator* this, PlayState* play) {
+    s32 i;
+    Vec3f ripplePos;
+    Vec3f splashPos;
+    s32 j;
+    f32 d1;
+    f32 d2;
+    f32 v1;
+    f32 v2;
+    f32 var_fs0;
+    s32 rn;
+
+    Matrix_Push();
+    Matrix_RotateYS(this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+
+    for (i = 0; i < ARRAY_COUNT(D_80B929D0) - 1; i++) {
+        v1 = D_80B929D0[i];
+        v2 = D_80B929D8[i];
+        d1 = D_80B929D0[i + 1] - D_80B929D0[i];
+        d2 = D_80B929D8[i + 1] - D_80B929D8[i];
+
+        for (j = 0; j < 7; j++) {
+            ripplePos.x = ((d1 * j) * (1.0f / 7.0f)) + v1;
+            ripplePos.y = this->waterSurfaceYPos;
+            ripplePos.z = ((d2 * j) * (1.0f / 7.0f)) + v2;
+
+            ripplePos.x += (Rand_ZeroOne() - 0.5f) * 20.0f;
+            ripplePos.z += (Rand_ZeroOne() - 0.5f) * 20.0f;
+            Matrix_MultVec3f(&ripplePos, &splashPos);
+            splashPos.x += this->dyna.actor.world.pos.x;
+            splashPos.z += this->dyna.actor.world.pos.z;
+            EffectSsGSplash_Spawn(play, &splashPos, NULL, NULL, 0, (Rand_ZeroOne() * 400.0f) + 210.0f);
+        }
+    }
+
+    for (i = 0; i < ARRAY_COUNT(sLargeRipplesLives); i++) {
+        ripplePos.x = ((Rand_ZeroOne() - 0.5f) * 60.0f) + this->dyna.actor.world.pos.x;
+        ripplePos.y = this->waterSurfaceYPos;
+        ripplePos.z = ((Rand_ZeroOne() - 0.5f) * 60.0f) + this->dyna.actor.world.pos.z;
+        EffectSsGRipple_Spawn(play, &ripplePos, 1000, 3000, sLargeRipplesLives[i]);
+    }
+
+    for (i = 0; i < ARRAY_COUNT(sSmallRipplesLives); i++) {
+        var_fs0 = Rand_ZeroOne();
+        var_fs0 = 1.0f - (var_fs0 * var_fs0);
+        rn = Rand_Next();
+        if (rn > 0) {
+            var_fs0 = -var_fs0;
+        }
+        ripplePos.x = (var_fs0 * 100.0f) + this->dyna.actor.world.pos.x;
+        ripplePos.y = this->waterSurfaceYPos;
+        var_fs0 = Rand_ZeroOne();
+        var_fs0 = 1.0f - (var_fs0 * var_fs0);
+        rn = Rand_Next();
+        if (rn > 0) {
+            var_fs0 = -var_fs0;
+        }
+        ripplePos.z = (var_fs0 * 100.0f) + this->dyna.actor.world.pos.z;
+        EffectSsGRipple_Spawn(play, &ripplePos, 400, 800, sSmallRipplesLives[i]);
+    }
+    Matrix_Pop();
+}
+
+s32 BgDblueElevator_GetWaterFlowFromCeiling(Actor* thisx, PlayState* play) {
+    BgDblueElevator* this = THIS;
+
+    if (Flags_GetSwitch(play, BG_DBLUE_ELEVATOR_GET_7F(&this->dyna.actor, 0))) {
+        return 0;
+    }
+    return 1;
+}
+
+s32 BgDblueElevator_GetWaterFlowFromPipes(Actor* thisx, PlayState* play) {
+    BgDblueElevator* this = THIS;
+    s32 var_s0 = 0;
```

Comment:
```suggestion
    s32 waterflow = BG_DBLUE_ELEVATOR_WATER_FLOW_STOPPED;
```
I think you could use the defines for this function (and the one above) for the return values. Note you would need to add a `#define BG_DBLUE_ELEVATOR_WATER_FLOW_FORWARD 1` for the use of the 1 below.